### PR TITLE
Use CronJob TPR to store state for 'last time job was run'

### PR DESCRIPTION
**Pull request by joshk0**
The prior implementation used the corpse of a completed Job to do so.

### CODE REVIEW

File: kubcron/managers/cronjobs.py [hunk 1]
```diff
@@ -2,27 +2,28 @@
...
 
     def watch(self):
         """Start listening to the kubernetes api and updating state."""
-        for message in self._watch('/apis/epicconstructions.com/v1alpha1/cronjobs'):
+        for message in self.k8s_watch(self.API_PREFIX):
             obj = message['object']
             name = '{}_{}'.format(obj['metadata']['namespace'], obj['metadata']['name'])
             if message['type'] in ['ADDED', 'MODIFIED']:
-                self.cronjobs[name] = self._clean(obj)
+                self.cronjobs[name] = obj
```
Comment (dclark87) [inline kubcron/managers/cronjobs.py]: So here - it's safe to have the un-cleaned metadata for the job in the manager's state? In other words, we only need it to be cleaned for the JobManager and not the CronJobManager?

File: kubcron/scheduler.py [hunk 2]
```diff
@@ -19,42 +19,64 @@ def __init__(self, cronjob_manager, job_manager, interval_secs=30):
...
+        job_spec = cronjob['spec']['jobTemplate']['spec']
+        job_metadata = job_spec.setdefault('template', {'metadata': {}})
+        job_labels = job_metadata.setdefault('labels', {})
 
-        spec['template']['metadata']['labels']['name'] = metadata['name']
+        job_labels['name'] = cronjob_metadata['name']
 
         # Set a default restartPolicy
-        spec['template']['spec']['restartPolicy'] = spec['template']['spec'].get('restartPolicy', 'OnFailure')
+        job_spec['template']['spec'].setdefault('restartPolicy', 'OnFailure')
```
Comment (dclark87) [inline kubcron/scheduler.py]: Nice gettin rid of the churn and noise.

File: kubcron/scheduler.py [hunk 3]
```diff
@@ -19,42 +19,64 @@ def __init__(self, cronjob_manager, job_manager, interval_secs=30):
...
+                    name, status['lastTrigger'], last_trigger_time))
+            else:
+                status['lastTrigger'] = last_trigger_time
+                logging.info('Triggering creation of daughter job {}'.format(name))
+                self._trigger_job(cronjob, name)
+                cronjob['status'] = status
+                logging.info('Triggering update of parent CronJob {}'.format(cronjob_name))
+                self._cronjob_manager.update(cronjob)
 
     def _job_exists(self, namespace, name):
```
Comment (dclark87) [inline kubcron/scheduler.py]: Nit: this method is no longer used and can be deleted.

File: kubcron/managers/cronjobs.py (hunk 1 above)
Comment (joshk0) [inline kubcron/managers/cronjobs.py]: Right. The cleaning requirement is when cloning the CronJob to create a granular Job for a specific run. We actually need the entire obj for the purpose of refreshing the last run time, posting the old metadata (specifically resourceVersion) is required to get it to update successfully. So, that's why i moved clean into the Job creator.

File: kubcron/scheduler.py (hunk 3 above)
Comment (joshk0) [inline kubcron/scheduler.py]: gone

**Pull request closed without merging**